pkg/reconciler/reconciler.go

```diff
@@ -0,0 +1,206 @@
+	WithInjectConfigFunc(InjectConfigFunc) ReconcilerBuilder
+	WithInjectClientFunc(InjectClientFunc) ReconcilerBuilder
+}
+
+type ReconcilerBuilderImpl struct {
+	r *reconciler
+}
+
+var _ ReconcilerBuilder = &ReconcilerBuilderImpl{}
+
+func (b *ReconcilerBuilderImpl) Build() reconcile.Reconciler {
+	return b.r
+}
+
+func (b *ReconcilerBuilderImpl) WithFinalizer(finalizerName string, finalizer FinalizerFunc) ReconcilerBuilder {
+	b.r.finalizerName = finalizerName
+	b.r.finalizer = finalizer
+	return b
+}
+
+func (b *ReconcilerBuilderImpl) WithFilter(filter FilterFunc) ReconcilerBuilder {
+	b.r.filter = filter
+	return b
+}
+
+func (b *ReconcilerBuilderImpl) WithLogger(logger *zap.Logger) ReconcilerBuilder {
+	b.r.logger = logger
+	return b
+}
+
+func (b *ReconcilerBuilderImpl) WithRecorder(recorder record.EventRecorder) ReconcilerBuilder {
+	b.r.recorder = recorder
+	return b
+}
+
+func (b *ReconcilerBuilderImpl) WithInjectConfigFunc(ic InjectConfigFunc) ReconcilerBuilder {
+	b.r.injectConfig = ic
+	return b
+}
+
+func (b *ReconcilerBuilderImpl) WithInjectClientFunc(ic InjectClientFunc) ReconcilerBuilder {
+	b.r.injectClient = ic
+	return b
+}
+
+func NewBuilder(er EventingReconciler) ReconcilerBuilder {
+	//	if i,ok := er.(client.Client)
+	return &ReconcilerBuilderImpl{r: &reconciler{EventingReconciler: er}}
+}
+
+type FilterFunc func(context.Context, ReconciledResource, record.EventRecorder) bool
+type FinalizerFunc func(context.Context, ReconciledResource, record.EventRecorder) error
+type InjectClientFunc func(client.Client) error
+type InjectConfigFunc func(*rest.Config) error
+
+type reconciler struct {
+	finalizerName string
+	client        client.Client
+	recorder      record.EventRecorder
+	logger        *zap.Logger
+	EventingReconciler
+	filter       FilterFunc
+	finalizer    FinalizerFunc
+	injectConfig InjectConfigFunc
+	injectClient InjectClientFunc
+}
+
+var _ reconcile.Reconciler = &reconciler{}
+var _ inject.Client = &reconciler{}
+var _ inject.Config = &reconciler{}
+
+// inject.Client impl
+func (r *reconciler) InjectClient(c client.Client) error {
+	r.client = c
+	if r.injectClient != nil {
+		return r.injectClient(c)
+	}
+	return nil
+}
+
+// inject.Config impl
+func (r *reconciler) InjectConfig(c *rest.Config) error {
+	if r.injectConfig != nil {
+		return r.injectConfig(c)
+	}
+	return nil
+}
+
+// reconcile.Reconciler impl
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	obj := r.GetNewReconcileObject()
+	recObjTypeName := reflect.TypeOf(obj).Name() //TODO: Investigate why .Name() returns "". Need Name and not namespace.name
+
+	// TODO: Fix the code for cases where logger, recorder, or client are not set.
+	ctx := logging.WithLogger(context.TODO(), r.logger.With(zap.Any("request", request)))
+
+	if r.filter != nil && !r.filter(ctx, obj, r.recorder) {
+		logging.FromContext(ctx).Debug(fmt.Sprintf("Not reconciling %s as ShouldReconcile() returned false", recObjTypeName))
```

Comment:
Don't need the fmt. What is being reconciled is already in the log's request attribute.

This looks to be true on most (all?) of the logging statements in this method.